spec/requests/authors_spec.rb

```diff
@@ -1,11 +1,25 @@
-require 'rails_helper'
+require "rails_helper"
 
-RSpec.describe "Authors", type: :request do
+RSpec.describe AuthorsController, type: :controller do
   describe "GET /show" do
-    it "returns http success" do
-      get "/authors/show"
-      expect(response).to have_http_status(:success)
+    context "when author id valid" do
```

Comment:
Tìm hiểu về `describe, context, it`

Reply:
Rspec sử dụng những từ như `describe` và `it` để thể hiện hành vi, tổ chức của codes giống như một cuộc hội thoại:
```
"Describe an order."
"It sums the prices of its line items."
```
- Phương thức describe tạo ra một example group. Bên trong describe block, có thể khai báo các nested groups sử dụng `describe` hoặc `context`, hoặc có thể khai báo các examples sử dụng `it`.
- `context` là một phương pháp tốt giúp cho tests của bạn trở nên rõ ràng, dễ đọc và có tổ chức tốt hơn.
**VD:**
Một group, một examples: 
```ruby
 RSpec.describe "something" do
   it "does something" do
   end
 end
``` 
Nested example group (sử dụng context):
```ruby
 RSpec.describe "something" do
   context "in one context" do
     it "does one thing" do
     end
   end
 end
```